app/parse.py

```diff
@@ -15,8 +28,60 @@ class Product:
     num_of_reviews: int
 
 
+def parse_single_product(product_soup: BeautifulSoup) -> Product:
+    return Product(
+        title=product_soup.select_one(".title")["title"],
+        description=product_soup.select_one(
+            ".description"
+        ).text.replace("\xa0", " "),
+        price=float(product_soup.select_one(".price").text.replace("$", "")),
+        rating=len(product_soup.select(".ws-icon.ws-icon-star")),
+        num_of_reviews=int(
+            product_soup.select_one("p.pull-right").text.split()[0]
+        )
+    )
+
+
+def more_button(url: str) -> BeautifulSoup:
+    driver = webdriver.Chrome()
+    driver.get(url)
+
+    try:
+        more = driver.find_element(By.CLASS_NAME, "ecomerce-items-scroll-more")
+    except NoSuchElementException:
+        more = None
+
+    while more:
+        style_value = more.get_attribute("style")
+        if "display: none;" in style_value:
```

Comment:
You can simplify it by using <button_soup_object>.is_displayed() and <button_soup_object>.click() methods

Reply:
agree